src/Engine.ts

```diff
@@ -61,6 +61,11 @@ export class Engine {
     const txValid = await tx.verify(this.chainTracker)
     if (!txValid) throw new Error('Unable to verify SPV information.')
 
+    // Broadcast the transaction
+    if (this.broadcaster !== undefined) {
+      await this.broadcaster.broadcast(tx)
```

Comment:
Does this throw an error if status returned is that the broadcast was unsuccessful? I'll need to run the double spend test again to check.

At a glance it doesn't check the status.

Reply:
Confirmed. There is more to this.

Repeating the submission of the same transaction results in success (as you'd want) with ARC broadcast. The de-duplication logic saves you from this, so no problem.

The remaining issue is that nothing is done with the ARC response - 
```json
{
  arcResponse: {
    status: 'success',
    txid: 'b06cb1059c2bf3debee1e65dc90d0815653325c7cf2584763b90c761c09919db',
    message: 'SEEN_IN_ORPHAN_MEMPOOL '
  }
}
```

This transaction is a double spend so perhaps consider ARC at fault (no I think it immediately calls back once it realizes what's happened). Anyway - orphan should not be considered success. So we ought to go back and make sure the ARC client is making good decisions about whether or not a tx is considered successfully broadcast.

Reply:
If ARC can correctly identify the double spend and return an error response, the code change I pushed will check the status and throw a new error if an error is detected.